source/api/commands/route2.md

```diff
@@ -0,0 +1,449 @@
+
+- **Expects**: `Fixture` | `Body`
+
+- **Interface**:
+
+```ts
+{
+  /**
+   * If set, serve a fixture as the response body.
+   */
+  fixture?: Fixture
+  /**
+   * If set, serve a static string/JSON object as the response body.
+   */
+  body?: Body
+  /**
+   * @default {}
+   */
+  headers?: { [key: string]: string }
+  /**
+   * @default 200
+   */
+  statusCode?: number
+  /**
+   * If `forceNetworkError` is truthy, Cypress will destroy the connection to the browser and send no response. Useful for simulating a server that is not reachable. Must not be set in combination with other options.
+   */
+  forceNetworkError?: boolean
+}
+```
+
+# Examples
+
+## Without Stubbing
+
+### Wait on `GET` request matching `url`
+
+```javascript
+cy.route2('**/users').as('getUsers')
+cy.visit('/users')
+cy.wait('@getUsers')
+```
+
+### Wait on matching `method` and `url`
+
+```javascript
+cy.route2('POST', '**/users').as('postUser')
+cy.visit('/users')
+cy.get('#first-name').type('Julius{enter}')
+cy.wait('@postUser')
+```
+
+### Match route against any UserId
+
+```javascript
+cy.route2('**/users/*/comments')
+
+// https://localhost:7777/users/123/comments     <-- matches
+// https://localhost:7777/users/123/comments/465 <-- does not match
+```
+
+### Use glob to match all segments
+
+```javascript
+cy.route2('**/posts/**')
+
+// https://localhost:7777/posts/1            <-- matches
+// https://localhost:7777/posts/foo/bar/baz  <-- matches
+// https://localhost:7777/posts/quuz?a=b&1=2 <-- matches
+```
+
+### Modify request
+
+Before a request gets sent to the actual endpoint, we can modify the request. The supplied `RouteHandler` callback receives the request object as a parameter.
+
+You can also (optionally) add a simulated {% urlHash 'delay' Simulate-delay %} or {% urlHash 'throttle' Simulate-throttle %} on the request to be sent out.
+
+```javascript
+cy.route2('/users', (req) => {
+  // modify the headers or body
+  req.headers.accept = 'application/json'
+  
+  // To modify a JSON response,
+  // it will be provided as a string 
+  // in req.body and must be
+  // parsed and stringified properly.
+  const requestBody = JSON.parse(req.body)
+
+  req.body = JSON.stringify({
+    ...requestBody, 
+    note: 'Custom note' 
+  })
+})
+```
+
+### Modify response
+
+After a request gets passed through, we can modify the response from the server. This is done by passing the `req.reply()` method a callback function which receives the original response (i.e., `res`) as the first argument.
+
+```js
+cy.route2('/users', (req) => {
+  // passing a function to req.reply causes the request to pass through
+  // and allows the response from the origin server to be modified
+  req.reply((res) => {
+    res.status = 200
+    res.headers['x-new-headers'] = 'from-server'
+
+    // dynamically alter body
+    res.body = res.body.replace('window.top', 'window.self')
+    // now, the response will continue to the browser
```

Comment:
so this is the only place where req.reply is mentioned...

but the res object doesn't seem to be documented. none of these functions are documented anywhere, even though they provide critical functionality to modifying origin responses:

https://github.com/cypress-io/cypress/blob/cc357f98992ace16d39bbfcdd155421c9d474b15/packages/net-stubbing/lib/external-types.ts#L19-L39

Reply:
ex. res.send, res.delay, res.throttle...